src/preact/component/intersection-observer-resize.js

```diff
@@ -3,25 +3,28 @@ import {
   unobserveWithSharedInOb,
 } from '#core/dom/layout/viewport-observer';
 
-import {useEffect} from '#preact';
+import {useCallback} from '#preact';
 
 /**
  * Uses a shared IntersectionObserver per window instance to observe the given `ref`.
  *
- * @param {{current: Element}} ref
  * @param {function(IntersectionObserverEntry)} callback
- * @param {?Window} targetWin
+ * @return {function}
  */
-export function useIntersectionObserver(ref, callback, targetWin) {
-  useEffect(() => {
-    const node = ref.current;
-    if (!node) {
-      return;
-    }
-    observeWithSharedInOb(node, callback);
+export function useIntersectionObserver(callback) {
+  const refCb = useCallback(
+    (node) => {
+      if (!node) {
+        return;
+      }
+      observeWithSharedInOb(node, callback);
```

Comment:
I sadly don't think ref callbacks accept cleanup functions so we need to make one manually.
See https://github.com/facebook/react/issues/15176

```suggestion
export function useIntersectionObserver(callback) {
  const nodeRef = useRef(null);
  const refCb = useCallback(
    (node) => {
      const prevNode = nodeRef.current;
      nodeRef.current = node;
      if (prevNode) {
        unobserveWithSharedInOb(nodeRef.current, callback);
      }
      if (!node) {
        return;
      }
      
      observeWithSharedInOb(node, callback);
```